### PR TITLE
feat: add color preference setting

### PR DESCRIPTION
**Pull request by qiweiii**
## Summary

Add color preference in settings, allow user to choose green up / red up.

Solve the issue: #1158 

## Quick changelog

- Added `colorPreference` in settings

## What's new

*Explain in details what this PR solve or improve. You can include visuals.*

- User can choose color preference in settings.
- Instead of a fixed value, color of profit/loss in candle, tables and charts can be got from settingsStore.
- No change in dark / light theme.
- No change in entry/exit signals colors.

https://github.com/freqtrade/frequi/assets/32790369/714c586d-0ba5-44f6-8561-b858a950d4dd

https://github.com/freqtrade/frequi/assets/32790369/0dc083ed-d9a7-4228-af30-942526e0b983

### CODE REVIEW

File: src/views/SettingsView.vue [hunk 1]
```diff
@@ -71,6 +79,10 @@ const openTradesOptions = [
   { value: OpenTradeVizOptions.asTitle, text: 'Show in title' },
   { value: OpenTradeVizOptions.noOpenTrades, text: "Don't show open trades in header" },
 ];
+const colorPreferenceOptions = [
+  { value: 'greenUp', text: 'Green Up/Red Down' },
+  { value: 'redUp', text: 'Green Down/Red Up' },
+];
```
Comment (xmatthias) [inline src/views/SettingsView.vue:85]: I don't really like this way to set this up - as it's pretty restrictive, and will only fill exactly one requirement, ignoring all other accessibility requirements.

a better way is to allow the user to select custom colors (a user may be red/green colorblind, so swapping up with down won't help that case).
We can still have a "swap" button for candles (will ONLY apply to candles) - but the actual color should be selectable independently.

**Review by xmatthias — COMMENTED** [review]
I don't really like this way to set this up - as it's pretty restrictive, and will only fill exactly one requirement, ignoring all other accessibility requirements.

a better way is to allow the user to select custom colors (a user may be red/green colorblind, so swapping up with down won't help that case).
We can still have a "swap" button for candles (will ONLY apply to candles) - but the actual color should be selectable independently.

File: src/components/charts/CandleChart.vue [hunk 2]
```diff
@@ -91,7 +80,23 @@ const props = defineProps({
     type: Object as () => ChartSliderPosition,
     default: () => undefined,
   },
+  colorPreference: { required: false, type: String, default: 'greenUp' },
 });
+
+// Candle Colors
+const colorProfit = props.colorPreference === 'greenUp' ? '#26A69A' : '#EF5350';
+const colorLoss = props.colorPreference === 'greenUp' ? '#EF5350' : '#26A69A';
```
Comment (xmatthias) [inline src/components/charts/CandleChart.vue:88]: Why is this using hardcoded colors, instead of the colorStore?

File: src/stores/settings.ts [hunk 3]
```diff
@@ -32,6 +32,9 @@ export const useSettingsStore = defineStore('uiSettings', {
       useHeikinAshiCandles: false,
       notifications: notificationDefaults,
       profitDistributionBins: 20,
+      colorPreference: 'greenUp',
+      colorProfit: '#12bb7b',
+      colorLoss: '#ef5350',
```
Comment (xmatthias) [inline src/stores/settings.ts:37]: i MAY would extract colors from the settings-store into a separate store - but i guess having it here can be fine, too.

File: src/components/charts/CandleChart.vue (hunk 2 above)
Comment (qiweiii) [reply src/components/charts/CandleChart.vue:88]: I found that the color of candles (`#26A69A` and `#EF5350`) are different from the ones in  `colorProfit: '#12bb7b', colorLoss: '#ef5350'`, so I did not use the color in store directly

**Comment by qiweiii** [comment]
> I don't really like this way to set this up - as it's pretty restrictive, and will only fill exactly one requirement, ignoring all other accessibility requirements.
> 
> a better way is to allow the user to select custom colors (a user may be red/green colorblind, so swapping up with down won't help that case). We can still have a "swap" button for candles (will ONLY apply to candles) - but the actual color should be selectable independently.
> 
> Overall though - great you took this up 👍

I agree it could be further improved to allow user choose colors, another simpler way is to add a few options like this (we can change radio buttons into a select): 
image

For `a "swap" button for candles (will ONLY apply to candles)` part, I think it will confuse users, I usually expect the color representing profit/loss is consistent across the app.

File: src/components/charts/CandleChart.vue (hunk 2 above)
Comment (xmatthias) [reply src/components/charts/CandleChart.vue:88]: That's correct - it should be different colors, really. using the profit/loss colors doesn't play well within the actual chart.

**Comment by xmatthias** [comment]
> For a "swap" button for candles (will ONLY apply to candles) part, I think it will confuse users, I usually expect the color representing profit/loss is consistent across the app.

i disagree with this.
The reason i can imagine to swap green/red candles is for short trades - so mentally, the green color points the "right" direction.
profit/ loss on the other hand shouldn't be swapped with this - and i've seen no other system allowing to invert this (also not binance, as you pointed to that).

If any, this should be optional, but i don't actually see that as beneficial / necessary.

**Review by xmatthias — APPROVED** [review]
I made some minor adjustments (like extracing this to a separate store - as it will allow us for more flexibility in the future) - but overall, it looks and works great.

Thanks a lot :+1:

**Pull request merged**